src/Models/Test/ModelInstantiatorTest.php

```diff
@@ -156,6 +156,55 @@ public function testArrayToDataObjectBadNestedObject(): void
         }
     }
 
+    /**
+     * Validates that relationships can be null if defined that way
+     * in the data object.
+     * @throws ValidationException
+     */
+    public function testArrayToDataObjectPermitsNullableRelationsWhenDefinedNullable(): void
+    {
+        $fake_has_one_data_object = $this->getTestDataObject();
+        $fake_has_many_data_object = [$this->getTestDataObject()];
+
+        $data_object = $this->getTestDataObject();
+        $data_object->fakeHasOneRelation = $fake_has_one_data_object;
+        $data_object->nullableFakeHasOneRelation = null;
+        $data_object->fakeHasManyRelation = $fake_has_many_data_object;
+
+        $fake_relationship_array = [
+            'pdgaNumber' => 4297,
+            'firstName' => 'Ken',
+            'lastName' => 'Climo',
+            'testProperty' => true,
+            'email' => 'champ@pdga.com',
+            'privacy' => true,
+            'birthDate' => '2020-01-01T00:00:00+00:00',
+        ];
+
+        $array = [
+            'pdgaNumber' => 4297,
+            'firstName' => 'Ken',
+            'lastName' => 'Climo',
+            'testProperty' => true,
+            'email' => 'champ@pdga.com',
+            'privacy' => true,
+            'birthDate' => '2020-01-01T00:00:00+00:00',
+            'fakeHasOneRelation' => $fake_relationship_array,
+            'nullableFakeHasOneRelation' => null,
+            'fakeHasManyRelation' => [$fake_relationship_array],
+        ];
+
+        $actual = $this->model_instantiator->arrayToDataObject(
+            $array,
+            ModelInstantiatorTestObject::class,
+        );
+
+        $this->assertEquals(
+            $data_object,
+            $actual
+        );
+    }
+
```

Comment:
Do we need a test for a non-nullable relationship that is defined in the incoming array as null?

Reply:
Good call; I've added this test